src/main/java/ListNode.java

```diff
@@ -0,0 +1,44 @@
+public class ListNode {
+    public ListNode next;
+    public int data;
+
+    public ListNode(int data) {
+        this.data = data;
+    }
+
+    public static ListNode reverseLinkedList(ListNode head) {
+        ListNode prev = null;
+        ListNode next = null;
+        ListNode cur = head;
+        while(cur != null) {
+            next = cur.next;
+            cur.next = prev;
+            prev = cur;
+            cur = next;
+        }
+        return prev;
+    }
+
+    public static ListNode mergeLinkedLists(ListNode listA, ListNode listB) {h
+        ListNode head = new ListNode();
+        mergeLinkedLists(head, listA, listB);
+        return head.next;
+    }
+
+    private static void mergeLinkedLists(ListNode merged, ListNode a, ListNode b) {
```

Comment:
If "merged" is your output, java does pass by value, so the result will never get saved. You need to return the merged list.

Reply:
Ah nevermind, you're passing in a new node and setting the pointer value on it.